zerver/actions/message_edit.py

```diff
@@ -436,7 +436,7 @@ def do_update_message(
     rendering_result: MessageRenderingResult | None,
     prior_mention_user_ids: set[int],
     mention_data: MentionData | None = None,
-) -> int:
+) -> tuple[int, list[dict[str, Any]]]:
```

Comment:
Can you make this return a dataclass with two properties? That'd be more readable.

Reply:
Created two new dataclasses, first time using them, seems pretty neat.